sdk/web-pubsub/web-pubsub/src/groupClient.ts

```diff
@@ -0,0 +1,305 @@
+ */
+export interface GroupSendTextToAllOptions extends OperationOptions {
+  /**
+   * Connection ids to exclude from receiving this message.
+   */
+  excludedConnections?: string[];
+  contentType: "text/plain";
+}
+
+/**
+ * Client for connecting to a Web PubSub group.
+ */
+export class WebPubSubGroup {
+  private client!: GeneratedClient;
+
+  /**
+   * The name of this group
+   */
+  public readonly groupName: string;
+
+  /**
+   * The name of the hub this group belongs to
+   */
+  public readonly hubName: string;
+
+  /**
+   * The Web PubSub API version being used by this client
+   */
+  public readonly apiVersion: string = "2020-10-01";
+
+  /**
+   * The Web PubSub endpoint this client is connected to
+   */
+  public endpoint!: string;
+
+  /**
+   * @private
+   * @internal
+   */
+  constructor(hubName: string, groupName: string) {
+    this.groupName = groupName;
+    this.hubName = hubName;
+  }
+  /**
+   * Add a specific connection to this group
+   *
+   * @param connectionId The connection id to add to this group
+   * @param options Additional options
+   */
+  public async addConnection(
+    connectionId: string,
+    options: GroupAddConnectionOptions = {}
+  ): Promise<RestResponse> {
+    const { span, updatedOptions } = createSpan(
+      "WebPubSubManagementClient-group-addConnection",
+      options
+    );
+
+    try {
+      const res = await this.client.webPubSub.addConnectionToGroup(
+        this.hubName,
+        this.groupName,
+        connectionId,
+        updatedOptions
+      );
+
+      if (res._response.status === 404) {
+        throw new RestError(
+          `Connection id '${connectionId}' doesn't exist`,
+          undefined,
+          res._response.status,
+          res._response.request,
+          res._response
+        );
+      }
+
+      return res;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Remove a specific connection from this group
+   *
+   * @param connectionId The connection id to remove from this group
+   * @param options Additional options
+   */
+  public async removeConnection(
+    connectionId: string,
+    options: GroupRemoveConnectionOptions = {}
+  ): Promise<RestResponse> {
+    const { span, updatedOptions } = createSpan(
+      "WebPubSubManagementClient-group-removeUser",
+      options
+    );
+
+    try {
+      const res = await this.client.webPubSub.removeConnectionFromGroup(
+        this.hubName,
+        this.groupName,
+        connectionId,
+        updatedOptions
+      );
+
+      return res;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Add a user to this group
+   *
+   * @param username The user name to add
+   * @param options Additional options
+   */
+  public addUser(username: string, options: GroupAddUserOptions = {}): Promise<RestResponse> {
+    const { span, updatedOptions } = createSpan("WebPubSubManagementClient-group-addUser", options);
+
+    try {
+      return this.client.webPubSub.addUserToGroup(
```

Comment:
Here and every other method you're wrapping in spans, I believe you need to await the operation and return that value. Otherwise, you'll call span.end() in the finally block before the operation has actually resolved.

```suggestion
      return await this.client.webPubSub.addUserToGroup(
```